pints/_mcmc/_slice_doubling.py

```diff
@@ -0,0 +1,559 @@
+#
+# Slice Sampling with Doubling MCMC Method
+#
+# This file is part of PINTS.
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing information, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+from __future__ import absolute_import, division
+from __future__ import print_function, unicode_literals
+import pints
+import numpy as np
+
+
+class SliceDoublingMCMC(pints.SingleChainMCMC):
```

Comment:
See the changes I made to the documentation for stepout. Think that we should try to make these similar here. Reading this does make me think that we could streamline slice samplers by creating a more abstract base class though...

Reply:
Thanks @ben18785 for the changes to the documentation, they look great! I will follow this example for the doubling documentation as well. Regarding creating a more abstract class for slice sampling, I will discuss the idea with Martin tomorrow.